cleanlab/datalab/issue_manager/duplicate.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, ClassVar, Dict, List, Optional, Union
+import warnings
+
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+from scipy.sparse import csr_matrix
+from sklearn.neighbors import NearestNeighbors
+from sklearn.utils.validation import check_is_fitted
+
+from cleanlab.datalab.issue_manager import IssueManager
+
+if TYPE_CHECKING:  # pragma: no cover
+    from cleanlab.datalab.datalab import Datalab
+
+
+class NearDuplicateIssueManager(IssueManager):
+    """Manages issues related to near-duplicate examples."""
+
+    description: ClassVar[
+        str
+    ] = """A (near) duplicate issue refers to two or more examples in
+    a dataset that are extremely similar to each other, relative
+    to the rest of the dataset.  The examples flagged with this issue
+    may be exactly duplicated, or lie atypically close together when
+    represented as vectors (i.e. feature embeddings).
+    """
+    issue_name: ClassVar[str] = "near_duplicate"
+    verbosity_levels = {
+        0: [],
+        1: ["threshold"],
+        2: [],
+    }
+
+    def __init__(
+        self,
+        datalab: Datalab,
+        metric: Optional[str] = None,
+        threshold: Optional[float] = None,
```

Comment:
what do you think about making this have the default value = 0.2 and applying it always as a multiplicative factor to the median-distances? I.e. instead of this being a distance-value (which users will be less sure how to tweak when they didn't compute the distances themselves), it is just a multiplier to adjust the median distance to then determine the actual distance-threshold to determine `is_issue`.

So I'm proposing: `is_issue[i] = distances[i] < threshold * np.median(distances)` always, and requiring user to provide threshold between 0 and 1, if they don't want default value.  Note that lower values of the threshold imply less issues detected

Reply:
Switched for duplicate.py.